lib/libutee/tee_api_arith_mpi.c

```diff
@@ -0,0 +1,837 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright (c) 2018, Linaro limited
+ */
+#include <assert.h>
+#include <mbedtls/bignum.h>
+#include <mempool.h>
+#include <stdio.h>
+#include <string.h>
+#include <util.h>
+#include <utee_defines.h>
+#include <utee_syscalls.h>
+#include <tee_api.h>
+#include <tee_arith_internal.h>
+
+#define MPI_MEMPOOL_SIZE (8 * 1024)
+
+static void __noreturn api_panic(const char *func, int line, const char *msg)
+{
+	printf("Panic function %s, line %d: %s\n", func, line, msg);
+	TEE_Panic(0xB16127 /*BIGINT*/);
+	while (1)
+		; /* Panic will crash the thread */
+}
+
+#define API_PANIC(x) api_panic(__func__, __LINE__, x)
+
+static void __noreturn mpi_panic(const char *func, int line, int rc)
+{
+	printf("Panic function %s, line %d, code %d\n", func, line, rc);
+	TEE_Panic(0xB16127 /*BIGINT*/);
+	while (1)
+		; /* Panic will crash the thread */
+}
+
+#define MPI_CHECK(x) do { \
+		int _rc = (x); \
+		 \
+		if (_rc) \
+			mpi_panic(__func__, __LINE__, _rc); \
+	} while (0)
+
+void _TEE_MathAPI_Init(void)
+{
+	static uint8_t data[MPI_MEMPOOL_SIZE] __aligned(__alignof__(long));
```

Comment:
This `long` relates to the 32bit arithmetic constraint described at few places right?

Reply:
This is copy/paste from `tee_api_arith_mpa.c`.
I'll change it to `mbedtls_mpi_uint`.